clients/src/main/java/org/apache/kafka/common/telemetry/internals/SinglePointMetric.java

```diff
@@ -0,0 +1,157 @@
+        this.key = key;
+        this.metricBuilder = metricBuilder;
+    }
+
+    @Override
+    public MetricKey key() {
+        return key;
+    }
+
+    public Metric.Builder builder() {
+        return metricBuilder;
+    }
+
+    public static SinglePointMetric create(MetricKey metricKey, Metric.Builder metric) {
+        return new SinglePointMetric(metricKey, metric);
+    }
+
+    /*
+        Methods to construct gauge metric type.
+     */
+    public static SinglePointMetric gauge(MetricKey metricKey, Number value, Instant timestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value);
+        return gauge(metricKey, point);
+    }
+
+    public static SinglePointMetric gauge(MetricKey metricKey, double value, Instant timestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value);
+        return gauge(metricKey, point);
+    }
+
+    /*
+        Methods to construct sum metric type.
+     */
+
+    public static SinglePointMetric sum(MetricKey metricKey, double value, boolean monotonic, Instant timestamp) {
+        return sum(metricKey, value, monotonic, timestamp, null);
+    }
+
+    public static SinglePointMetric sum(MetricKey metricKey, double value, boolean monotonic, Instant timestamp,
+        Instant startTimestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value);
+        if (startTimestamp != null) {
+            point.setStartTimeUnixNano(toTimeUnixNanos(startTimestamp));
+        }
+
+        return sum(metricKey, AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE, monotonic, point);
+    }
+
+    public static SinglePointMetric deltaSum(MetricKey metricKey, double value, boolean monotonic,
+        Instant timestamp, Instant startTimestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value)
+            .setStartTimeUnixNano(toTimeUnixNanos(startTimestamp));
+
+        return sum(metricKey, AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA, monotonic, point);
+    }
+
+    /*
+        Helper methods to support metric construction.
+     */
+    private static SinglePointMetric sum(MetricKey metricKey, AggregationTemporality aggregationTemporality,
+        boolean monotonic, NumberDataPoint.Builder point) {
+        Objects.requireNonNull(point, "metric number data point cannot be null");
+        Objects.requireNonNull(metricKey, "metric key cannot be null");
+
+        point.addAllAttributes(asAttributes(metricKey.tags()));
+
+        Metric.Builder metric = Metric.newBuilder().setName(metricKey.getName());
+        metric
+            .getSumBuilder()
+            .setAggregationTemporality(aggregationTemporality)
+            .setIsMonotonic(monotonic)
+            .addDataPoints(point);
+        return create(metricKey, metric);
+    }
+
+    private static SinglePointMetric gauge(MetricKey metricKey, NumberDataPoint.Builder point) {
+        Objects.requireNonNull(point, "metric number data point cannot be null");
```

Comment:
Do we need this check?

Reply:
Can drop, not adding much value. Removed it.